### ISSUE TITLE
dqlite patch failed when tried to build in custom

### ISSUE THREAD

**Issue by hvoid-build-block** [ubuntu/microk8s#1897]
Please run `microk8s inspect` and attach the generated tarball to this issue.
I tried to set up the lxc and build with a custom version of k8s with the following cmd:
```shell
lxc exec test-build -- sh -c "cd microk8s && SNAPCRAFT_BUILD_ENVIRONMENT=host KUBE_VERSION=1.19.x snapcraft"
```
where x = 0 -> 7

The build failed at chaging `etcd` to `storage`. The running script was `build-k8s-binaries.sh`  Here is the log:
```shell
+ cd /root/microk8s/parts/k8s-binaries/build/go/src/github.com/kubernetes/kubernetes
+ git config user.email microk8s-builder-bot@ubuntu.com
+ git config user.name MicroK8s builder bot
+ PATCHES=patches
+ echo v1.19.1
+ grep -e beta -e rc -e alpha
+ echo Applying patch /root/microk8s/build-scripts/patches/dqlite.patch
Applying patch /root/microk8s/build-scripts/patches/dqlite.patch
+ git am
Applying: Rename flag set "etcd" to "storage"
error: patch failed: cmd/kube-apiserver/app/options/options.go:167
error: cmd/kube-apiserver/app/options/options.go: patch does not apply
Patch failed at 0001 Rename flag set "etcd" to "storage"
The copy of the patch that failed is found in: .git/rebase-apply/patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
Failed to run 'override-build': Exit code was 128.
```

What to do in order to fix this ?

Thank you~~

**Comment by ktsakalozos** [comment]
Hi @hvoid-build-block. You could use the dqlite.patch that is available on the 1.19 branch [1].

The patches are extracted from [2] with `git format-patch <start-commit>^.<end-commit>` .

Interested to hear why you do your own build. 

[1] https://github.com/ubuntu/microk8s/tree/1.19/build-scripts/patches  
[2] https://github.com/canonical/kubernetes-dqlite/tree/release-1.19

**Comment by hvoid-build-block** [comment]
Thank you for the solution~
